static/sidebar/js/mobile/modal/index.js

```diff
@@ -28,6 +28,19 @@
     document.body.appendChild(script);
   };
 
+  var checkProducts = function(data){
+    var relatedProds = Utils.getByClass('tvp-related-products');
+    var productsHolder = Utils.getByClass('tvp-products');
+    if (!data.length) {
+      relatedProds.style.display = 'none';
+      productsHolder.style.display = 'none';
+    }else{
+      relatedProds.style.display = '';
+      productsHolder.style.display = '';
+    }
+  };
+
+
```

Comment:
I think there's an easier way to solve this, without involving much code, please see below comment